src/sys/statfs.rs

```diff
@@ -1,22 +1,432 @@
-use {Result, NixPath};
+use {NixPath, Result};
 use errno::Errno;
-use std::os::unix::io::AsRawFd;
 use libc;
+use std::mem;
+use std::os::unix::io::AsRawFd;
 
-pub fn statfs<P: ?Sized + NixPath>(path: &P, stat: &mut libc::statfs) -> Result<()> {
-    unsafe {
-        Errno::clear();
-        let res = try!(
-            path.with_nix_path(|path| libc::statfs(path.as_ptr(), stat))
-        );
+#[cfg(target_os = "android")]
+pub type fsid_t = libc::__fsid_t;
+#[cfg(not(target_os = "android"))]
+pub type fsid_t = libc::fsid_t;
+
+#[derive(Clone, Copy)]
+#[allow(missing_debug_implementations)]
+pub struct Statfs(libc::statfs);
```

Comment:
It would be very handy to have Debug for this structure.  You could implement it manually.  See SigEvent in sys/signal.rs for an example.  I wouldn't hold up the PR just for that, though.